sycl/source/detail/graph_impl.cpp

```diff
@@ -1784,6 +1828,34 @@ std::vector<node> modifiable_command_graph::get_root_nodes() const {
   return createNodesFromImpls(Impls);
 }
 
+void modifiable_command_graph::checkNodePropertiesAndThrow(
```

Comment:
Does this use any member variables of the `modifiable_command_graph` class? If we could make it a static free function like `static void checkGraphPropertiesAndThrow()` then it would avoid making the function part of the ABI (and so avoid any risk of ABI breaking changes). We would need to move the definition of the `modifiable_command_graph::add()` functions from `graph.hpp` to `graph_impl.cpp`, but I think that's okay.

Reply:
no, it is not using any members, I could easily make it static. movement of ::add methods will not work since one of them is templated: 
`template <typename T> node add(T CGF, const property_list &PropList = {})`

Reply:
ah thanks for checking about the `add` methods, making the method `static` would be explicit that we're not using any members so think that's a good change.